IntegrationTests/TimersTest.js

```diff
@@ -111,62 +119,77 @@ const TimersTest = createReactClass({
       0,
     );
     let failAnim; // This should fail without the t=0 fastpath feature.
-    this.setTimeout(() => this.cancelAnimationFrame(failAnim), 0);
-    failAnim = this.requestAnimationFrame(() =>
+    this._timeoutID = setTimeout(() => cancelAnimationFrame(failAnim), 0);
+    failAnim = requestAnimationFrame(() =>
       this._fail(
         'testOrdering-Anim, setTimeout 0 should happen before ' +
           'requestAnimationFrame',
       ),
     );
     let fail25;
-    this.setTimeout(() => {
-      this.clearTimeout(fail25);
+    this._timeoutID = setTimeout(() => {
+      clearTimeout(fail25);
     }, 20);
-    fail25 = this.setTimeout(
+    fail25 = setTimeout(
       () =>
         this._fail(
           'testOrdering-t25, setTimeout 20 should happen before ' +
             'setTimeout 25',
         ),
       25,
     );
-    this.setTimeout(this.done, 50);
-  },
+    this._timeoutID = setTimeout(this.done, 50);
+  }
 
   done() {
     this.setState({done: true}, () => {
       TestModule.markTestCompleted();
     });
-  },
+  }
+
+  componentWillUnmount() {
+    if (this._timeoutID != null) {
+      clearTimeout(this._timeoutID);
+    }
+    if (this._immediateId != null) {
+      clearImmediate(this._immediateId);
+    }
+    if (this._intervalId != null) {
+      clearInterval(this._intervalId);
+    }
+  }
 
   render() {
     return (
       <View style={styles.container}>
         <Text>
-          {this.constructor.displayName + ': \n'}
+          {/* $FlowFixMe(>=0.54.0 site=react_native_fb,react_native_oss) This
+             * comment suppresses an error found when Flow v0.54 was deployed.
+             * To see the error delete this comment and run Flow. */
+          this.constructor.displayName + ': \n'}
           Intervals: {this.state.count + '\n'}
           {this.state.done ? 'Done' : 'Testing...'}
         </Text>
       </View>
     );
-  },
+  }
 
   _incrementInterval() {
     if (this.state.count > 3) {
       throw new Error('interval incremented past end.');
     }
     if (this.state.count === 3) {
-      this.clearInterval(this._interval);
+      clearInterval(this._intervalId);
       this.setState({count: 0}, this._nextTest);
       return;
     }
     this.setState({count: this.state.count + 1});
-  },
+  }
 
   _fail(caller: string): void {
     throw new Error('_fail called by ' + caller);
-  },
-});
+  }
+}
 
 const styles = StyleSheet.create({
   container: {
```

Comment:
Let's get rid of: 
```JavaScript
TimersTest.displayName = 'TimersTest';
```

If we want to define static properties on `TimerTest`, we should do it using the `static` keyword inside the `TimerTest` class. That way, flow will be able to analyze the static property.